addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -169,24 +180,68 @@ def get_mail_channel(self, livechat_channel_id, anonymous_name, user_id=None, co
         mail_channel._broadcast([operator_partner_id])
         return mail_channel.sudo().channel_info()[0]
 
+    def _get_random_operator(self):
+        """ Return a random operator from the available users of the channel that have the lowest number of active livechats.
+        A livechat is considered 'active' if it has at least one message within the 30 minutes.
+
+        (Some annoying conversions have to be made on the fly because this model holds 'res.users' as available operators
+        and the mail_channel model stores the partner_id of the randomly selected operator)
+
+        Returns:
+            user_id (res.users): a random operator
```

Comment:
Indeed you return a user, not a user_id ;)

Reply:
Fixette.